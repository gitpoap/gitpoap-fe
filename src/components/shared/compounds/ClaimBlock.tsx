import React, { useEffect, useState } from 'react';
import { rem, tint } from 'polished';
import { Button } from '../elements/Button';
import { PrimaryBlue } from '../../../colors';
import { GitPOAP } from './GitPOAP';
import { FaCheckCircle, FaCoins, FaEthereum } from 'react-icons/fa';
import { useReward } from 'react-rewards';
import { Badge, Stack, Tooltip } from '@mantine/core';

type Props = {
  gitPOAPId: number;
  imgSrc: string;
  name: string;
  orgName?: string;
  description: string;
  onClickClaim: () => void;
  onClickBadge?: () => void;
  isClaimingAll: boolean;
  isClaimed?: boolean;
  isLoading?: boolean;
  isConnected?: boolean;
  issuedVia?: 'github' | 'email' | 'eth';
};

const getButtonText = (isClaimed: boolean | undefined, isConnected: boolean | undefined) => {
  if (isClaimed) {
    return 'Minted';
  }

  if (!isConnected) {
    return 'Connect';
  }

  return 'Mint';
};

export const ClaimBlock = ({
  gitPOAPId,
  imgSrc,
  name,
  orgName,
  description,
  onClickClaim,
  onClickBadge,
  isClaimed,
  isLoading,
  isClaimingAll,
  isConnected,
  issuedVia,
}: Props) => {
  const [isClaimedPrev, setIsClaimedPrev] = useState<boolean>(!!isClaimed);
  const rewardId = 'rewardId-' + gitPOAPId;
  const { reward } = useReward(rewardId, 'confetti', {
    colors: [
      tint(1.0, PrimaryBlue),
      tint(0.8, PrimaryBlue),
      tint(0.6, PrimaryBlue),
      tint(0.4, PrimaryBlue),
      tint(0.2, PrimaryBlue),
      tint(0.0, PrimaryBlue),
    ],
    elementCount: 100,
    spread: 60,
  });

  useEffect(() => {
    if (isClaimed && !isClaimedPrev) {
      reward();
      setIsClaimedPrev(true);
    }
  }, [isClaimed, isClaimedPrev, reward]);

  return (
    <Stack align="center">
      <GitPOAP
        gitPOAPId={gitPOAPId}
        imgSrc={imgSrc}
        name={name}
        repoName={orgName}
        description={description}
        onClick={onClickBadge}
      />
      {issuedVia && (
        <Tooltip label={`Issued via ${issuedVia}`}>
          <Badge size="sm" variant="filled" style={{ letterSpacing: rem(1) }}>
            {issuedVia}
          </Badge>
        </Tooltip>
      )}
      <Button
        id={rewardId}
        onClick={onClickClaim}
        loading={isLoading}
        leftIcon={isClaimed ? <FaCheckCircle /> : !isConnected ? <FaEthereum /> : <FaCoins />}
        disabled={isClaimed || isClaimingAll}
      >
        {getButtonText(isClaimed, isConnected)}
      </Button>
    </Stack>
  );
};
