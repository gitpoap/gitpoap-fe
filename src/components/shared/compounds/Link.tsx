import styled from 'styled-components';
import NextLink from 'next/link';
import { IconStyles, IconStylesHover } from '../elements/icons';
import { trackLinkClick } from '../../../lib/tracking/events';

type Props = React.ComponentProps<typeof NextLink>;

export const Link = (props: Props) => {
  const { children, className, ...restProps } = props;

  return (
    <NextLink
      {...restProps}
      className={className}
      target={restProps.target}
      rel={restProps.rel}
      onClick={(e) => {
        trackLinkClick(restProps.href.toString(), e.target);
        if (restProps.onClick) {
          restProps.onClick(e);
        }
      }}
    >
      {children}
    </NextLink>
  );
};

export const IconLink = styled(Link)`
  ${IconStyles}
  ${IconStylesHover}
`;
