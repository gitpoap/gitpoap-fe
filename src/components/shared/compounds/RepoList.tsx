import styled from 'styled-components';
import { rem } from 'polished';
import { BREAKPOINTS } from '../../../constants';

export const RepoList = styled.div`
  display: grid;
  width: 100%;
  margin-top: ${rem(32)};
  margin-bottom: ${rem(32)};

  justify-content: center;
  align-content: center;
  align-items: flex-start;

  grid-template-columns: repeat(auto-fit, minmax(${rem(300)}, 1fr));
  column-gap: ${rem(24)};
  row-gap: ${rem(24)};
  @media (max-width: ${rem(1000)}) {
    column-gap: ${rem(16)};
    row-gap: ${rem(16)};
  }
`;

export const RepoListSmall = styled.div`
  display: grid;
  width: 100%;
  margin-top: ${rem(50)};
  margin-bottom: ${rem(55)};

  grid-template-columns: repeat(auto-fit, ${rem(200)});
  column-gap: ${rem(40)};
  row-gap: ${rem(35)};

  @media (max-width: ${rem(750)}) {
    column-gap: 5%;
    grid-template-columns: repeat(auto-fit, 30%);
  }

  @media (max-width: ${BREAKPOINTS.sm}px) {
    column-gap: 4%;
    grid-template-columns: repeat(auto-fit, 48%);
  }
`;
