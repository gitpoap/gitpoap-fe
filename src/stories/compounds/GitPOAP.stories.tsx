import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import { GitPOAP } from '../../components/shared/compounds/GitPOAP';
import { gitPOAPs } from '../data';

export default {
  title: 'Compounds/GitPOAP',
  component: GitPOAP,
} as ComponentMeta<typeof GitPOAP>;

const Template: ComponentStory<typeof GitPOAP> = (args) => {
  return <GitPOAP {...args} />;
};

export const Default = Template.bind({});
Default.args = gitPOAPs[0];

export const NoDescription = Template.bind({});
NoDescription.args = {
  id: gitPOAPs[0].id,
  imgSrc: gitPOAPs[0].imgSrc,
  name: gitPOAPs[0].name,
  orgName: gitPOAPs[0].orgName,
};
