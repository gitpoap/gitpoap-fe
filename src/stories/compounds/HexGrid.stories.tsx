import React from 'react';
import styled from 'styled-components';
import { rem } from 'polished';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import { GitPOAPBadge } from '../../components/shared/elements/GitPOAPBadge';

import real_badge1 from '../assets/gitPOAPs/real_badge1.png';
import real_badge2 from '../assets/gitPOAPs/real_badge2.png';
import real_badge3 from '../assets/gitPOAPs/real_badge3.png';
import real_badge4 from '../assets/gitPOAPs/real_badge4.png';
import real_badge5 from '../assets/gitPOAPs/real_badge5.png';
import real_badge6 from '../assets/gitPOAPs/real_badge6.png';
import real_badge7 from '../assets/gitPOAPs/real_badge7.png';
import real_badge8 from '../assets/gitPOAPs/real_badge8.png';
import real_badge9 from '../assets/gitPOAPs/real_badge9.png';
import real_badge10 from '../assets/gitPOAPs/real_badge10.png';
import real_badge11 from '../assets/gitPOAPs/real_badge11.png';
import real_badge12 from '../assets/gitPOAPs/real_badge12.png';

const Main = styled.div`
  display: flex;
  --s: 100px; /* size  */
  --m: 4px; /* margin */
  --f: calc(var(--s) * 1.732 + 4 * var(--m) - 1px);
  width: ${rem(1350)};
`;

const Container = styled.div`
  font-size: 0; /* disable white space between inline block element */

  &::before {
    content: '';
    width: calc(var(--s) / 2 + var(--m));
    float: left;
    height: 120%;
    shape-outside: repeating-linear-gradient(#0000 0 calc(var(--f) - 3px), #000 0 var(--f));
  }
`;

const Hex = styled.div<{ featured: boolean; imgUrl?: string }>`
  width: var(--s);
  margin: var(--m);
  height: calc(var(--s) * 1.1547);
  display: inline-block;
  font-size: initial;
  clip-path: polygon(0% 25%, 0% 75%, 50% 100%, 100% 75%, 100% 25%, 50% 0%);
  margin-bottom: calc(var(--m) - var(--s) * 0.2885);

  text-align: center;
  line-height: calc(var(--s) * 1.1547);

  background: ${(props) => (props.featured ? 'red' : 'green')};
  // background: ${(props) =>
    props.imgUrl ? `no-repeat center / 100% url('${props.imgUrl}')` : 'blue'};
  //   &:nth-child(odd) {
  //     background: green;
  //   }
`;

const StyledGitPOAPBadge = styled(GitPOAPBadge)`
  position: absolute;
`;

let width = 12;
let height = 4;

let gitpoaps = [
  real_badge1,
  real_badge2,
  real_badge3,
  real_badge4,
  real_badge5,
  real_badge6,
  real_badge7,
  // real_badge8,
  // real_badge9,
  // real_badge10,
  // real_badge11,
  // real_badge12,
].slice(0, 7);

// https://css-tricks.com/hexagons-and-beyond-flexible-responsive-grid-patterns-sans-media-queries/
const HexGrid = ({}) => {
  let length = width * height;

  let middle = Math.floor(gitpoaps.length / 2);

  let top = gitpoaps.length > 3 ? gitpoaps.slice(0, middle) : gitpoaps;
  let bottom = gitpoaps.length > 3 ? gitpoaps.slice(middle) : [];

  let topStart = Math.floor(12 + width / 2 - top.length / 2);
  let bottomStart = Math.ceil(24 + width / 2 - bottom.length / 2);

  const map = new Map();
  top.map((item, i) => map.set(topStart + i, item));
  bottom.map((item, i) => map.set(bottomStart + i, item));

  return (
    <Main>
      <Container>
        {[...Array(length)].map((_, i) => {
          return <Hex key={i} featured={map.has(i)} />;
          // return (
          //   <Hex key={i} featured={map.has(i)}>
          //     {map.has(i) ? <StyledGitPOAPBadge imgUrl={map.get(i)} size="sm" /> : null}
          //   </Hex>
          // );
        })}
      </Container>
    </Main>
  );
};

export default {
  title: 'Compounds/HexGrid',
  component: HexGrid,
} as ComponentMeta<typeof HexGrid>;

const Template: ComponentStory<typeof HexGrid> = (args) => {
  return <HexGrid {...args} />;
};

export const Default = Template.bind({});
