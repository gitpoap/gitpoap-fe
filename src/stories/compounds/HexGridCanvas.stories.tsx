import React, { useEffect, useRef, useState } from 'react';
import styled from 'styled-components';
import { rem } from 'polished';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import { GitPOAP } from '../../components/shared/compounds/GitPOAP';
import { gitPOAPs } from '../data';
import { GitPOAPBadge } from '../../components/shared/elements/GitPOAPBadge';

import real_badge1 from '../assets/gitPOAPs/real_badge1.png';
import real_badge2 from '../assets/gitPOAPs/real_badge2.png';
import real_badge3 from '../assets/gitPOAPs/real_badge3.png';
import real_badge4 from '../assets/gitPOAPs/real_badge4.png';
import real_badge5 from '../assets/gitPOAPs/real_badge5.png';
import real_badge6 from '../assets/gitPOAPs/real_badge6.png';
import real_badge7 from '../assets/gitPOAPs/real_badge7.png';
import real_badge8 from '../assets/gitPOAPs/real_badge8.png';
import real_badge9 from '../assets/gitPOAPs/real_badge9.png';
import real_badge10 from '../assets/gitPOAPs/real_badge10.png';
import real_badge11 from '../assets/gitPOAPs/real_badge11.png';
import real_badge12 from '../assets/gitPOAPs/real_badge12.png';

const Main = styled.div`
  display: flex;
  --s: 100px; /* size  */
  --m: 4px; /* margin */
  --f: calc(var(--s) * 1.732 + 4 * var(--m) - 1px);
  width: ${rem(1350)};
`;

const Container = styled.div`
  font-size: 0; /* disable white space between inline block element */

  &::before {
    content: '';
    width: calc(var(--s) / 2 + var(--m));
    float: left;
    height: 120%;
    shape-outside: repeating-linear-gradient(#0000 0 calc(var(--f) - 3px), #000 0 var(--f));
  }
`;

const Hex = styled.div<{ featured: boolean; imgUrl?: string }>`
  width: var(--s);
  margin: var(--m);
  height: calc(var(--s) * 1.1547);
  display: inline-block;
  font-size: initial;
  clip-path: polygon(0% 25%, 0% 75%, 50% 100%, 100% 75%, 100% 25%, 50% 0%);
  margin-bottom: calc(var(--m) - var(--s) * 0.2885);

  text-align: center;
  line-height: calc(var(--s) * 1.1547);

  //   background: ${(props) => (props.featured ? 'red' : 'green')};
  background: ${(props) =>
    props.imgUrl ? `no-repeat center / 100% url('${props.imgUrl}')` : 'blue'};
  //   &:nth-child(odd) {
  //     background: green;
  //   }
`;

const StyledGitPOAPBadge = styled(GitPOAPBadge)`
  position: absolute;
`;

let width = 12;
let height = 4;
// let gitpoaps = [...Array(5)];
let gitpoaps = [
  real_badge1,
  real_badge2,
  real_badge3,
  real_badge4,
  real_badge5,
  real_badge6,
  real_badge7,
  real_badge8,
  real_badge9,
  real_badge10,
  real_badge11,
  real_badge12,
].slice(0, 7);

let images = [
  'https://assets.poap.xyz/gitpoap-2020-truffle-contributor-2020-logo-1649262289456.png',
  'https://assets.poap.xyz/gitpoap-2018-web3js-contributor-2018-logo-1649267123396.png',
  'https://assets.poap.xyz/gitpoap-2019-truffle-contributor-2019-logo-1649262236424.png',
  'https://assets.poap.xyz/gitpoap-2020-trufflesuitecom-contributor-2020-logo-1649268807709.png',
  'https://assets.poap.xyz/gitpoap-2022-trufflesuitecom-contributor-2022-logo-1649268861179.png',
  'https://assets.poap.xyz/gitpoap-2017-truffle-contributor-2017-logo-1649262094154.png',
  'https://assets.poap.xyz/gitpoap-2017-trufflesuitecom-contributor-2017-logo-1649268651751.png',
  'https://assets.poap.xyz/gitpoap-2022-truffle-contributor-2022-logo-1649262354204.png',
  'https://assets.poap.xyz/gitpoap-2018-truffle-contributor-2018-logo-1649262185269.png',
  'https://assets.poap.xyz/gitpoap-2021-truffle-contributor-2021-logo-1649262333030.png',
];

// https://eperezcosano.github.io/hex-grid/
// https://stackoverflow.com/questions/43911775/how-to-draw-an-infinite-hexagon-spiral
const HexGrid = ({}) => {
  const canvasRef = useRef<HTMLCanvasElement | null>(null); //document.getElementById('canvas');
  const [canvas, setCanvas] = useState<HTMLCanvasElement | null>(null);
  const [ctx, setCtx] = useState<CanvasRenderingContext2D | null>(null);

  const a = (2 * Math.PI) / 6;
  const r = 100;

  useEffect(() => {
    if (canvasRef) {
      setCanvas(canvasRef.current);
    }
  }, [canvasRef]);

  useEffect(() => {
    if (canvas) {
      setCtx(canvas.getContext('2d'));
    }
  }, [canvas]);

  useEffect(() => {
    if (ctx) {
      cluster(canvas.width / 2, canvas.height / 2, images.length);
    }
  }, [ctx]);

  const hexagon = (centerX: number, centerY: number, it: number) => {
    // console.log(centerX, centerY, it);
    // var x = centerX + Math.cos((Math.PI * 2) / 6) * r;
    // var y = centerY + Math.sin((Math.PI * 2) / 6) * r;

    // var x = centerX + r * Math.cos(0);
    // var y = centerY + r * Math.sin(0);
    let rs = r - 2;

    const image = new Image();
    image.src = images[it % images.length];
    image.onload = () => {
      ctx.save();
      ctx.beginPath();
      //   ctx.moveTo(x, y);
      ctx.moveTo(
        centerX + Math.cos(0 - (Math.PI * 2) / 4) * r,
        centerY + Math.sin(0 - (Math.PI * 2) / 4) * r,
      );

      for (var i = 1; i < 7; i++) {
        // x = centerX + Math.cos(((Math.PI * 2) / 6) * i) * r;
        // y = centerY + Math.sin(((Math.PI * 2) / 6) * i) * r;

        // ctx.lineTo(x, y);
        ctx.lineTo(
          centerX + Math.cos((i * (Math.PI * 2)) / 6 - (Math.PI * 2) / 4) * r,
          centerY + Math.sin((i * (Math.PI * 2)) / 6 - (Math.PI * 2) / 4) * r,
        );
      }

      ctx.closePath();
      ctx.clip();

      ctx.drawImage(image, centerX - rs, centerY - rs, rs * 2, rs * 2); //, this.canvasA.width, this.canvasA.height);

      ctx.stroke();
      ctx.restore();
    };
  };

  const cluster = (centerX: number, centerY: number, count: number) => {
    var x = centerX,
      y = centerY,
      angle = Math.PI / 3,
      dist = Math.sin(angle) * (r * 2),
      side = 0;

    var it = 0;
    hexagon(x, y, it++);
    while (it < count) {
      for (var t = 0; t < Math.floor((side + 4) / 6) + (side % 6 == 0) && it < count; t++) {
        y = y - dist * Math.cos(side * angle - Math.PI / 2);
        x = x - dist * Math.sin(side * angle - Math.PI / 2);
        hexagon(x, y, it++);
      }
      side++;
    }
  };

  return <canvas ref={canvasRef} id="canvas" width="1200" height="800" />;
};

export default {
  title: 'Compounds/HexGridCanvas',
  component: HexGrid,
} as ComponentMeta<typeof HexGrid>;

const Template: ComponentStory<typeof HexGrid> = (args) => {
  return <HexGrid {...args} />;
};

export const Default = Template.bind({});

//   drawGrid(canvas.width, canvas.height);

//   function drawGrid(width: number, height: number) {
//     let i = 0;
//     for (let y = r; y + r * Math.sin(a) < height; y += r * Math.sin(a)) {
//       for (
//         let x = r, j = 0;
//         x + r * (1 + Math.cos(a)) < width;
//         x += r * (1 + Math.cos(a)), y += (-1) ** j++ * r * Math.sin(a)
//       ) {
//         drawHexagon(x, y, i);
//         i++;
//       }
//     }
//   }

//   function drawHexagon(x: number, y: number, i: number) {
//     const image = new Image();
//     image.src = images[i % images.length];
//     image.onload = () => {
//       ctx.save();
//       ctx.beginPath();
//       for (let i = 0; i < 6; i++) {
//         ctx.lineTo(x + r * Math.cos(a * i), y + r * Math.sin(a * i));
//       }
//       // ctx.drawImage(img, 0, 0);
//       // ctx.fillStyle = `#${Math.floor(Math.random() * 16777215).toString(16)}`;
//       // ctx.fill();
//       ctx.closePath();
//       ctx.clip();

//       ctx.drawImage(image, x - r, y - r, r * 2, r * 2); //, this.canvasA.width, this.canvasA.height);

//       ctx.stroke();
//       ctx.restore();
//     };

//   let r = 50;
