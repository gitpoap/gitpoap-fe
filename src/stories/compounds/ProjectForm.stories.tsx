import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import { ProjectForm } from '../../components/ProjectForm';

export default {
  title: 'Compounds/ProjectForm',
  component: ProjectForm,
} as ComponentMeta<typeof ProjectForm>;

const Template: ComponentStory<typeof ProjectForm> = () => {
  return <ProjectForm />;
};

export const Default = Template.bind({});
Default.args = {};
