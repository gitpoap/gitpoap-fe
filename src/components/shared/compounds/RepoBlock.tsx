import React from 'react';
import styled from 'styled-components';
import { rem } from 'polished';
import { BackgroundPanel, BackgroundPanel2, ExtraHover, ExtraPressed } from '../../../colors';
import { Body, InfoHexBase } from '../elements/InfoHexBase';
import { GitPOAP, People, Star } from '../elements/icons';
import { Text } from '../elements/Text';
import { TitleNoHover } from '../elements/Title';
import { useRepoStarCountQuery } from '../../../graphql/generated-gql';
import { textEllipses } from '../styles';
import { Group, Paper, Stack } from '@mantine/core';
import { useRouter } from 'next/router';
import { Link } from './Link';
import { TextSkeleton } from '../elements';

const RepoName = styled(TitleNoHover)`
  text-align: center;
  font-size: ${rem(18)};
  line-height: ${rem(22)};
  margin-bottom: ${rem(-2)};
  ${textEllipses(260)}
`;

const OrgName = styled(Text)`
  text-align: center;
  ${textEllipses(260)}

  &:hover {
    color: ${ExtraHover};
  }
  &:active {
    color: ${ExtraPressed};
  }
`;

const BODY_HEIGHT = 20;

export const InfoHexBaseStyled = styled(InfoHexBase)`
  ${Body} {
    display: flex;
    align-items: center;
    justify-content: center;

    margin-top: ${rem(BODY_HEIGHT)};
    margin-bottom: ${rem(BODY_HEIGHT)};

    &:before {
      height: ${rem(BODY_HEIGHT)};
      top: ${rem(-BODY_HEIGHT)};
    }
    &:after {
      height: ${rem(BODY_HEIGHT)};
      bottom: ${rem(-BODY_HEIGHT)};
    }

    &:hover {
      ${RepoName} {
        color: ${ExtraHover};
      }
    }

    &:active {
      ${RepoName} {
        color: ${ExtraPressed};
      }
    }
  }
`;

export const RepoBlockSkeleton = ({ className }: { className?: string }) => {
  return (
    <Paper
      className={className}
      sx={{
        background: BackgroundPanel,
        border: `1px solid ${BackgroundPanel}`,
        borderRadius: rem(6),
        transition: `150ms ease-in-out`,
        '&:hover': {
          background: BackgroundPanel2,
          borderColor: 'white',
          cursor: 'pointer',
        },
      }}
    >
      <Stack align="center" py={rem(24)} px={rem(20)} spacing={12} className={className}>
        <RepoName>
          <TextSkeleton height={rem(26)} />
        </RepoName>
        <OrgName>
          <TextSkeleton height={rem(18)} width={rem(140)} />
        </OrgName>
        <Group position="center">
          <TextSkeleton height={rem(18)} width={rem(140)} />
        </Group>
      </Stack>
    </Paper>
  );
};

type Repo = {
  id: number;
  name: string;
  contributorCount?: number;
  mintedGitPOAPCount?: number;
  organization: {
    name: string;
  };
};

type Props = {
  repo: Repo;
  className?: string;
  starredCount?: number;
  totalContributors?: number;
};

export const RepoBlock = ({ className, repo }: Props) => {
  // @TODO: Add back claimCount, uniqueContributorCount once backend changes are in
  const { id, name, contributorCount, mintedGitPOAPCount, organization } = repo;

  const router = useRouter();
  const [resultStarCount] = useRepoStarCountQuery({ variables: { repoId: id } });
  const starCount = resultStarCount?.data?.repoStarCount;

  return (
    <Paper
      className={className}
      sx={{
        background: BackgroundPanel,
        border: `1px solid ${BackgroundPanel}`,
        borderRadius: rem(6),
        transition: `150ms ease-in-out`,
        '&:hover': {
          background: BackgroundPanel2,
          borderColor: 'white',
          cursor: 'pointer',
        },
      }}
      onClick={() => router.push(`/gh/${repo.organization.name}/${repo.name}`)}
    >
      <Stack align="center" py={rem(24)} px={rem(20)} spacing={12} className={className}>
        <RepoName>{name}</RepoName>
        <Link href={`/gh/${repo.organization.name}`} passHref>
          <OrgName>{organization.name}</OrgName>
        </Link>
        {/* @TODO: Add back once backend changes are ready - Jay (Jul 3 2022) */}
        <Group position="center">
          <Group spacing={4}>
            <People />
            <Text>{contributorCount ?? 0}</Text>
          </Group>
          <Group spacing={4}>
            <GitPOAP />
            <Text>{mintedGitPOAPCount ?? 0}</Text>
          </Group>
          <Group spacing={4}>
            <Star />
            <Text>{starCount ?? 0}</Text>
          </Group>
        </Group>
      </Stack>
    </Paper>
  );
};
