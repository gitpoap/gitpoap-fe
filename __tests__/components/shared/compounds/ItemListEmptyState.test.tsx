import { render, screen } from '@testing-library/react';
import 'jest-styled-components';
import { EmptyState } from '../../../../src/components/shared/compounds/ItemListEmptyState';

describe('ItemListEmptyState', () => {
  it('should render a icon', () => {
    render(
      <EmptyState icon={<img src="/test/test.png" />}>
        <div>test</div>
      </EmptyState>,
    );
    const icon = screen.getByRole('img');
    expect(icon).toBeInTheDocument();
    expect(icon).toMatchSnapshot();
  });

  it('should render a children', () => {
    render(
      <EmptyState icon={<img src="/test/test.png" />}>
        <div>test1</div>
        <div>test2</div>
      </EmptyState>,
    );
    const test1 = screen.getByText('test1');
    expect(test1).toBeInTheDocument();
    expect(test1).toMatchSnapshot();
    const test2 = screen.getByText('test2');
    expect(test2).toBeInTheDocument();
    expect(test2).toMatchSnapshot();
  });
});
