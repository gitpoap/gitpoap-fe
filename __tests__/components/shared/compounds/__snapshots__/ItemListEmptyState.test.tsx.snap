// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemListEmptyState should render a children 1`] = `
<div
  class="mantine-Stack-root mantine-njf2rt"
  style="padding: 3.75rem; flex: 1; grid-column-end: 1; grid-row-end: 2;"
>
  <img
    src="/test/test.png"
  />
  <div>
    test1
  </div>
  <div>
    test2
  </div>
</div>
`;

exports[`ItemListEmptyState should render a icon 1`] = `
<img
  src="/test/test.png"
/>
`;
